report-generator/src/main/java/com/teamscale/report/ReportUtils.java

```diff
@@ -18,36 +19,42 @@
 /** Utilities for generating reports. */
 public class ReportUtils {
 
-	private static final Gson GSON = new GsonBuilder().setPrettyPrinting().create();
+	private static Moshi moshi = new Moshi.Builder().build();
 
 	/** Converts to given testwise coverage report to a json report and writes it to the given file. */
 	public static <T> void writeReportToFile(File reportFile, T report) throws IOException {
 		File directory = reportFile.getParentFile();
 		if (!directory.isDirectory() && !directory.mkdirs()) {
 			throw new IOException("Failed to create directory " + directory.getAbsolutePath());
 		}
-		try (FileWriter writer = new FileWriter(reportFile)) {
-			GSON.toJson(report, writer);
+		try (BufferedSink writer = Okio.buffer(Okio.sink(reportFile))) {
```

Comment:
Do we still want to call that "writer"? It does not look like it behaves like a typical Writer anymore.